.github/contrib/install.php

```diff
@@ -33,5 +33,9 @@
 if (file_exists('pre-commit')) {
     copy('pre-commit', '.git/hooks/pre-commit');
     chmod('.git/hooks/pre-commit', 0750);
-    echo "\e[32mPre-commit Hook has been installed.\e[0;m\n";
+} else {
+    file_put_contents('.git/hooks/pre-commit', file_get_contents(__DIR__.'/pre-commit'), FILE_APPEND | LOCK_EX);
```

Comment:
It seems you do the same thing actually. But with a wrong check: "if the file does not exist, get its content".

Perhaps `if (file_exists(__DIR__.'/pre-commit')) {` was enough?

Reply:
What do you think about my update?

Reply:
Ah better, now I understand we do not deal with the same file.
But ... If you make a backup of the old file, do you want to copy your new file in all cases? The `else` shouldn't be removed?